Fix running WebAssembly Browser App from VS

thaystg opened the pull request:
As `$(OutputPath)` is not absolute path, we need to add `$(MSBuildProjectDirectory)`

Tested:

- Create a new WebAssembly Browser App on VS and run, change WasmAppDir and run again.
- Create a new WebAssembly Console App on VS and run, change WasmAppDir and run again.
- Create a new wasmbrowser app using command line and run;
- Create a new wasmconsole app using command line and run;
- Create a new wasmbrowser app using command line and dotnet build and dotnet run --no-build;
- Create a new wasmbrowser app using command line and dotnet build and dotnet run --debug;

Also, added tests for command line runs. But a few of the tests are disabled because the tests would be using 7.0 targets, but 7.0 doesn't have the fix yet.

Fix #75356

thaystg committed:
trying to fix 75356

thaystg committed:
Trying to fix for running from VS and from command line

thaystg committed:
Remove unused spaces

thaystg committed:
Fix again

radical commented on src/mono/wasm/build/WasmApp.targets:
```diff
@@ -117,10 +117,7 @@
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == '' and Exists('$(OutputPath)/AppBundle')">$([MSBuild]::NormalizeDirectory($(OutputPath), 'AppBundle'))</_AppBundleDirForRunCommand>
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == '' and Exists('$(OutputPath)/browser-wasm/AppBundle')">$([MSBuild]::NormalizeDirectory($(OutputPath), 'browser-wasm', 'AppBundle'))</_AppBundleDirForRunCommand>
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == ''">OutputPath=$(OutputPath), OutDir=$(OutDir)</_AppBundleDirForRunCommand>
+    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == ''">$([MSBuild]::NormalizeDirectory($(MSBuildProjectDirectory), $(OutputPath), 'browser-wasm', 'AppBundle'))</_AppBundleDirForRunCommand>
```
We should check that `OutputPath` is not rooted, and only then assign prepend this.

thaystg committed:
Addressing @radical comments

radical committed:
Merge remote-tracking branch 'origin/main' into thays_fix_75356

radical committed:
OutputPath can be missing the rid during the evaluation

radical committed:
more cleanup

thaystg committed:
Replacing NormalizeDirectory with  [System.IO.Path]::Combine

radical committed:
[wasm] Add tests for running templates outside project directory

radical committed:
Merge remote-tracking branch 'origin/main' into thays_fix_75356

radical committed:
Add tests for browser template

radical commented:
/backport to release/7.0

radical committed:
Disable failing test

.. which is failing because the WBT is still using net7.0 on `main`,
thus using the 7.0 targets, which don't have the fix yet.

We can enable this once the fix is merged for 7.0 .

radical commented:
The corresponding 7.0 PR has been merged. Once the packages from that are available, the disabled test here can be enabled again.

lewing commented:
failure is https://github.com/dotnet/runtime/issues/74302
